application/use_cases/products/create_product/create_product_input_port.py

```diff
@@ -1,13 +1,11 @@
 from clapy import InputPort
 
-from domain.entities.base_entity import EntityID
-
 
 class CreateProductInputPort(InputPort):
     brand: str = None
     image: bytes = None
     is_available: bool
-    merchant_id: EntityID
+    merchant_name: str
```

Comment:
curious as to why by name now?

Reply:
So had this realisation when I was redoing the merchant API. Originally I had put Dora in control of merchants and defining which ones existed, when really it should be up to the merchant API to say which merchants it supports. Dora should only upset merchants as they come across from the merchant API.

Also I originally thought I'd be setting up the merchants as system data in the seed file, but that also doesn't seem necessary. So now it's an upsert, but we can change it if something else makes more sense 😄

Reply:
That makes sense & sounds like a good change.

You could give Dora Merchants their own CRUD & MAPI could interact with that to create a Dora Merchant when it doesn't exist. To then send in the Dora Merchant ID when creating a Product BUT regardless i think that is unnecessary complexity. Especially considering Merchants are NOT user defined.